pkg/virt-controller/watch/vm.go

```diff
@@ -2714,40 +2713,67 @@ func (c *VMController) handleDynamicInterfaceRequests(vm *virtv1.VirtualMachine,
 		return nil
 	}
 
-	interfaceMap := map[string]virtv1.Interface{}
-	if vmi != nil {
-		interfaceMap = netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
-	}
 	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
-	for i, ifaceRequest := range vm.Status.InterfaceRequests {
+	for i, _ := range vm.Status.InterfaceRequests {
 		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
 			vmTemplateCopy, &vm.Status.InterfaceRequests[i])
 
 		if vmi == nil || vmi.DeletionTimestamp != nil {
 			continue
 		}
 
-		if ifaceRequest.AddInterfaceOptions != nil {
-			if _, exists := interfaceMap[ifaceRequest.AddInterfaceOptions.Name]; !exists {
-				if err := c.clientset.VirtualMachineInstance(vmi.Namespace).AddInterface(context.Background(), vmi.Name, ifaceRequest.AddInterfaceOptions); err != nil {
-					return err
-				}
-			}
-		}
+		vmiSpecCopy := vmi.Spec.DeepCopy()
```

Comment:
Why a copy is used?

Reply:
Since `c.VMIInterfacesPatch`compares between the changed copy and the original VMI to know if there is a need to patch the VMI.